e2e_tests/camino/cchain_nomock.test.ts

```diff
@@ -0,0 +1,584 @@
+        return {}
+      }
+    ],
+    [
+      "importKey of user",
+      () => xChain.importKey(user, passwd, adminPrivateKey),
+      (x) => x,
+      Matcher.toBe,
+      () => adminAddress
+    ],
+    [
+      "send tokens to gas fee address",
+      () =>
+        xChain.send(
+          user,
+          passwd,
+          "CAM",
+            900000000000,
+          gasFeeXAddr,
+          [adminAddress],
+            adminAddress,
+          "memo"
+        ),
+      (x) => x.txID,
+      Matcher.Get,
+      () => tx,
+      3000
+    ],
+    [
+      "export X->C gas fee address",
+      () =>
+        (async function () {
+          const avmUTXOResponse: GetUTXOsResponse = await xChain.getUTXOs(
+            gasFeeXAddr
+          )
+          const utxoSet: avm.UTXOSet = avmUTXOResponse.utxos
+
+          const unsignedTx: avm.UnsignedTx = await xChain.buildExportTx(
+            utxoSet,
+            new BN(800000000000),
+            avalanche.getNetwork().C.blockchainID,
+            [cAddressStrings[1]],
+            [gasFeeXAddr],
+            [gasFeeXAddr]
+          )
+
+          const tx: avm.Tx = unsignedTx.sign(xChain.keyChain())
+          return xChain.issueTx(tx)
+        })(),
+      (x) => x,
+      Matcher.Get,
+      () => tx,
+      3000
+    ],
+    [
+      "import X->C gas fee address",
+      () =>
+        (async function () {
+          const baseFeeResponse: string = await cChain.getBaseFee()
+          const baseFee = new BN(parseInt(baseFeeResponse, 16) / 1e9)
+          const evmUTXOResponse: any = await cChain.getUTXOs(
+            cAddressStrings,
+            avalanche.getNetwork().X.blockchainID
+          )
+          const utxoSet: UTXOSet = evmUTXOResponse.utxos
+          const avaxAssetID: Buffer = await cChain.getAVAXAssetID()
+
+          let unsignedTx: UnsignedTx = await cChain.buildImportTx(
+            utxoSet,
+            gasFeeAddr,
+            [cAddressStrings[1]],
+            avalanche.getNetwork().X.blockchainID,
+            cChain.getDefaultTxFee(),
+          )
+          const importCost: number = costImportTx(
+            avalanche.getNetwork().C,
+            unsignedTx
+          )
+          const fee = baseFee.mul(new BN(importCost))
+
+          unsignedTx = await cChain.buildImportTx(
+            utxoSet,
+            gasFeeAddr,
+            [cAddressStrings[1]],
+            avalanche.getNetwork().X.blockchainID,
+            fee,
+          )
+
+          const tx: Tx = unsignedTx.sign(cChain.keyChain())
+          return cChain.issueTx(tx)
+        })(),
+      (x) => x,
+      Matcher.Get,
+      () => tx,
+      3000
+    ],
+    [
+      "verify import tx has been committed",
+      () => cChain.getAtomicTxStatus(tx.value),
+      (x) => x,
+      Matcher.toBe,
+      () => "Accepted",
+      3000
+    ],
+    // Role Addition
+    [
+      "grant gas fee role to gasFeeAddr",
+      () =>
+        contract.methods
+          .grantRole(gasFeeAddr, 2)
```

Comment:
I would suggest to include all role constants in a separate file under /src/apis/evm (maybe camino_contstants.ts or something like that) and reference all roles in this test via static constants